src/NonceTracker.ts

```diff
@@ -208,14 +208,22 @@ export class NonceTracker {
     // we need to make sure our base count
     // and pending count are from the same block
     const blockNumber: string = await this.blockTracker.getLatestBlock();
-    const baseCount: number = await this.web3.getTransactionCount(
-      address,
-      blockNumber,
-    );
-    assert(
-      Number.isInteger(baseCount),
-      `nonce-tracker - baseCount is not an integer - got: (${typeof baseCount}) "${baseCount}"`,
-    );
+    const baseCountAsHex = await new Promise<string>((resolve, reject) => {
+      this.ethQuery.sendAsync<string[], string>(
```

Comment:
Nit: We could drop `ethQuery` here and call the provider directly:

```suggestion
      this.provider.sendAsync<string[], string>(
```